components/builder-api/src/server/services/github.rs

```diff
@@ -408,3 +414,28 @@ async fn read_plan_targets(github: &GitHubClient,
 
     Ok(targets)
 }
+
+fn suppress_autobuild_for_origin(config: &Config, origin: &str) -> bool {
+    config.api.suppress_autobuild_origins.contains(&origin.to_string())
+}
+
+#[cfg(test)]
+mod test {
+    use crate::config::Config;
+    use crate::server::services::github::suppress_autobuild_for_origin;
+
+    #[test]
+    fn suppress_autobuild_verify() {
```

Comment:
I am not clear on how we can create a dummy Project and pass it in to build_plans.  If we passed a Project in, then we are now building a project, not the plans.  We would move the plans iterator out of this function, then it would become build_plan.
Please clarify.

Reply:
ahh right. That iterator complicates this. Well you could still extract more logic into something like:

```
fn should_reject_build(plan: &PlanWithTarget, project: Project, config: &Config, repo_url: &str) -> bool {
    if repo_url != project.vcs_data {
        warn!("Repo URL ({}) doesn't match project vcs data ({}). Aborting.",
              repo_url, project.vcs_data);
        return true;
    }

    if !config.api.build_targets.contains(&plan.1) {
        debug!("Rejecting build with target: {:?}", plan.1);
        return true;
    }

    if config.api.suppress_autobuild_origins.contains(&plan.0.origin.to_string()) {
        debug!("Skipping autobuild for origin {:?}", &plan.0.origin);
        return true;
    }
    return false
}
```

And here would be an example "happy path" test:

```
    #[test]
    fn valid_plan_is_not_rejected() {
        let mut config = Config::default();
        config.api.suppress_autobuild_origins = vec!["hest".to_string(), "sest".to_string()];

        let project = Project { id: 1,
            owner_id: 1,
            origin: "origin".to_string(),
            package_name: "pkg".to_string(),
            name: "origin/pkg".to_string(),
            plan_path: "plan.sh".to_string(),
            target: "x86_64-linux".to_string(),
            vcs_type: "git".to_string(),
            vcs_data: "https://github.com/habitat-sh/testapp.git".to_string(),
            vcs_installation_id: None,
            auto_build: true,
            created_at: None,
            updated_at: None };

        let plan = PlanWithTarget(Plan { name: "pkg".to_string(),
            origin: "origin".to_string(),
            version: None},
            PackageTarget::from_str("x86_64-linux").unwrap());

        assert_eq!(should_reject_build(&plan, project, &config, "https://github.com/habitat-sh/testapp.git"), false);
    }
```

I'd leave it up to you if you wanted to add something like that. I'm just not sure if the test as it stands in the PR is worth including.

Reply:
Yeah this makes sense.  We can encapsulate the failure and test each separately.  I think it is good to have coverage for all of these so I'll go ahead and add tests for the repo_url and build_targets check.